docs/design-docs/08-package-variant.md

```diff
@@ -0,0 +1,450 @@
+upstream package reference. When that is done, the package variant controller
+will update any existing Draft package under its ownership by doing the
+equivalent of a `kpt pkg update` to rebase the downstream on the new upstream
+revision. If a Draft does not exist, then the package variant controller will
+create a new Draft based on the current published downstream, and apply the `kpt
+pkg update` to rebase that. This updated Draft must then be proposed and
+approved like any other package change.
+
+If a floating tag is used[^notimplemented], then explicit modification of the `PackageVariant` is
+not needed. Rather, when the floating tag is moved to a new tagged revision of
+the upstream package, the package revision controller will notice and
+automatically propose and update to that revision. For example, the upstream
+package author may designate three floating tags: stable, beta, and alpha. The
+upstream package author can move these tags to specific revisions, and any
+PackageVariant resource tracking them will propose updates to their downstream
+packages.
+
+
+
+### Adoption and Deletion Policies
+When a `PackageVariant` resource is created, it will have a particular
+repository and package name as the downstream. The adoption policy controls
+whether the package variant controller takes over an existing package with that
+name, in that repository.
+
+Analogously, when a `PackageVariant` resource is deleted, a decision must be
+made about whether or not to delete the downstream package. This is controlled
+by the deletion policy.
+
+### Deletion Policy
+
+## Fan Out of Variant Generation
+
+TODO(johnbelamaric): Describe `PackageVariantSet`.
+
+## Example Use Cases
+
+- describe scenarios and the `PackageVariant`, `PackageVariantSet` resources that
+  would solve the scenarios
+
+### Automatically Organizational Customization of an External Package
+We can use a `PackageVariant` to provide basic changes that are needed when
+importing a package from an external repository. For example, suppose we have
+our own internal registry from which we pull all our images. When we import an
+upstream package, we want to modify any images listed in the upstream package to
+point to our registry. Additionally, we have a policy that all Pods must have a
+`chargeback-code` label, so we want to add that as a validating policy to the
+package.
+
+In a manual CLI-based workflow, we would accomplish this as follows.
+...
+
+With `PackageVariant`, we instead create the following resources:
+...
+
+### Creating a Namespace Per Tenant
+
+### Customizing A Workload By Region
+
+### Customizing A Workload By Cluster and Region
+
+### Customizing A Workload By Environment
+
+## Detailed Design
+
+### `PackageVariant` API
+
+#### Basic API
+
+- Upstream
+- Downstream
+- Annotations
+- Labels
+- Adoption and Deletion Policy
+
+#### Package Context Injection
+
+#### KRM Function Pipeline
+
+#### Configuration Injection
```

Comment:
@natasha41575 @mortent this section could use a close eye...

Reply:
I had a few questions, but overall I feel it is pretty well described. It is a somewhat complicated mechanism, so the examples are very helpful.